scipy/stats/_continuous_distns.py

```diff
@@ -3582,56 +3583,890 @@ class levy_stable_gen(rv_continuous):
+        if param=="C": 
+            alpha, beta, scale, loc = _param_switch("C", "A", alpha=alpha, delta=asymmetry, scale=scale, loc=loc)
+        if param=="P": 
+            alpha, beta, scale, loc = _param_switch("P", "A", alpha=alpha, p1=asymmetry, scale=scale, loc=loc)
+        
+        # generate the random vector of the given size self._size
         sz = self._size
-        alpha = broadcast_to(alpha, sz)
-        beta = broadcast_to(beta, sz)
-        TH = uniform.rvs(loc=-np.pi/2.0, scale=np.pi, size=sz,
+        U = uniform.rvs(loc=-np.pi / 2.0, scale=np.pi, size=sz,
                          random_state=self._random_state)
-        W = expon.rvs(size=sz, random_state=self._random_state)
-        aTH = alpha*TH
-        bTH = beta*TH
-        cosTH = np.cos(TH)
-        tanTH = np.tan(TH)
-        res = _lazywhere(alpha == 1,
-                         (alpha, beta, TH, aTH, bTH, cosTH, tanTH, W),
-                         alpha1func, f2=alphanot1func)
-        return res
+        E = expon.rvs(size=sz, random_state=self._random_state)
+        
+
+        def generate_stable_rvs(alpha, beta, U, E):
+            cos_U = np.cos(U)
+            tan_U = np.tan(U)
+            if alpha==1:
+                return (2. / np.pi) * ((np.pi / 2. + beta * U) * tan_U -
+                    beta * np.log(E * cos_U) / (np.pi / 2. + beta * U))
+            else:
+                tan_betaB = beta * np.tan(np.pi * alpha / 2.)
+                b = np.arctan(tan_betaB) / alpha
+
+                factor1 = np.sin(alpha * (U + b)) / cos_U**(1.0 / alpha)
+                factor2 = (np.cos((1. - alpha) * U - alpha * b) / E)**((1. - alpha) / alpha)
+                scaler = (1. + tan_betaB**2.)**(1 / (2. * alpha))
+                return scaler * factor1 * factor2
+        
+        return loc + (generate_stable_rvs(alpha, beta, U, E) / scale**(1. / alpha)
 
-    def _argcheck(self, alpha, beta):
-        return (alpha > 0) & (alpha <= 2) & (beta <= 1) & (beta >= -1)
 
+    @staticmethod
+    def _cf(k, alpha, beta, location=0, scale=1, parameterisation_type="A"):
+        """
+        Characteristic functions under different parameterizations.
+        Default parameterization is "A". 
+
+        Args:
+        ----------
+        k :  float
+        alpha : float
+            Stability parameter with values in (0,2]. 
+        beta : float
+            Asymmetry parameter under parameterization A, takes values in [-1, 1].
+        location: float
+            Location parameter. Default setting: location=0.
+        scale: float
+            Scale parameter under parameterization A, takes positive real values.
+            Default setting: scale=1. 
+        """
+
+        K = lambda a: a - 1. + np.sign(1. - a)
+
+        common_exponent = 1j * k * location - (np.abs(scale * k) ** alpha)
+        if parameterisation_type == "A":
+            Phi = lambda alpha, k: np.tan(np.pi * alpha / 2.) if alpha != 1 else -2.0*np.log(np.abs(k)) / np.pi
```

Comment:
no need to use `lambda` here (same below)

Reply:
Agreed. I've changed them to functions.

I would prefer not to take the code out of the functions (if that is what you meant), because it is easy to lose track of the variables being manipulated. Additionally, the names of these functions allow easy cross-reference if somebody is verifying this code against a paper. We could also take these functions out as class methods if we need to use them elsewhere in the future.

Let me know if this is discouraged. :)